internal/internal_search_attributes.go

```diff
@@ -287,19 +287,13 @@ func NewSearchAttributes(attributes ...SearchAttributeUpdate) SearchAttributes {
 	for _, attr := range attributes {
 		attr(&sa)
 	}
-	// remove nil values
-	for key, value := range sa.untypedValue {
```

Comment:
This check also didn't totally exclude nulls from ending up in `untypedValue ` anyway.